shpc/main/modules/base.py

```diff
@@ -236,14 +236,14 @@ def docgen(self, module_name, registry=None, out=None, branch="main"):
         """
         Render documentation for a module within a local registry.
         """
-        config = self._load_container(module_name)
+        config = self.get_module(module_name).config
 
         out = out or sys.stdout
         aliases = config.get_aliases()
         template = self.template.load("docs.md")
         registry = registry or defaults.github_url
-        github_url = "%s/blob/%s/%s/container.yaml" % (registry, branch, module_name)
-        registry_bare = registry.split(".com")[-1]
+        github_url = "%s/blob/%s/%s/container.yaml" % (registry, branch, config.name)
+        registry_bare = registry.split(".com/")[-1]
```

Comment:
Doesn't this always assume we have a .com with a trailing slash? What if we kept the same but then did a final strip of `/`?

Reply:
Yes it assumes there is always a `/` after `.com`, because as far as I could see, `registry` is always a complete URL, even the [default value](https://github.com/singularityhub/singularity-hpc/blob/main/shpc/defaults.py#L24).
That said, and to make this bit future-proof (not all registry may be hosted on a `.com` domain), let's use `urllib` to extract the path from the URL ?

Reply:
Yes I think that's a fantastic idea - indeed custom GitLab registries don't always end in .com!